### DISCUSSION TITLE
flask-socket  How does flask-socket work across files?

### DISCUSSION THREAD

**Discussion by liuqijie6** [miguelgrinberg/Flask-SocketIO#1874] [Q&A]
This is a problem I had yesterday, I'm trying to solve it, hoping to get help here.

https://stackoverflow.com/questions/73633577/flask-socketio-cannot-print-in-factory-mode

**Comment by miguelgrinberg** [comment]
Based on the code you show in the question, the problem is that you are not importing test.py anywhere.

**Reply by liuqijie6** [reply]
Oh~ thanks, I really didn't know I needed to import socket_io in that file again.
